### ISSUE TITLE
Section title for Side menu

### ISSUE THREAD

**Issue by Spark450** [GovAlta/ui-components#1252]
## Acceptance Criteria:

- "Section title" is an available "menu item" type
- Section titles are non-interactive and help to organize menu items into sections
- [Figma design](https://www.figma.com/file/k9ZHctiEnN74GEhGWpio3K/Component---Side-Menu?type=design&node-id=2238%3A9253&t=ep61tqt2q9V9pa6U-1)
- Screen reader should announce the section heading once when the user navigates a set of child links.

### Accessibility

- Section titles **could** be implemented with styled `<h>` tags eg. `<h1>`. This should provide the accessibility we need for a section header. Note: this is one possible approach to ensure proper accessibility. Other options may be explored if they provide a better experience.
<img width="489" alt="Screen Shot 2023-06-27 at 12 16 17 PM" src="https://github.com/GovAlta/ui-components/assets/51723535/78d697a2-eafa-456b-a6c6-fff836a81ae4">
